crates/ruma-state-res/src/lib.rs

```diff
@@ -617,20 +651,25 @@ impl StateResolution {
     }
 
     /// Uses the `event_map` to return the full PDU or fails.
-    fn get_or_load_event<E: Event>(
+    fn get_or_load_event<E, F>(
```

Comment:
Do we even need this function anymore? Logs can be done inside of the closure

Reply:
Yeah, I mostly left it because it made it easy to refactor, but I removed it.